src/conn/conn_prefetch.c

```diff
@@ -105,6 +105,13 @@ __wt_prefetch_thread_run(WT_SESSION_IMPL *session, WT_THREAD *thread)
         if (pe == NULL)
             break;
 
+        /*
+         * Add this dhandle to the list of dhandles prefetch threads are currently working on. We
+         * use WT_PUBLISH here to guarantee write_once semantics on prefetch_dhandles. We don't
+         * require its ordering constraints.
+         */
+        WT_PUBLISH(conn->prefetch_dhandles[thread->id], pe->dhandle);
```

Comment:
The `thread->id` value could be zero, which could be the reason behind the problem of why the fix is not working.

Reply:
You can use the similar`btree->evict_busy` logic that is already existing as this value is incremented by the eviction threads. You can also use `btree->prefetch_busy` or `evict_busy` by renaming it whenever a prefetch thread is operating on that dhandle?

This way, there is no need for the array to store the dhandles.

Reply:
Thanks for the callout and idea! Attempted this approach with a new `btree->prefetch_busy` and it worked with no issues. Sticking to this approach. [171c916](https://github.com/wiredtiger/wiredtiger/pull/10189/commits/171c9163bd206848c37da30145477d4695e7a376)